server/routes/checkTransferController.ts

```diff
@@ -16,8 +17,24 @@ export default class CheckTransferController {
   public addToRoll(): RequestHandler {
     return async (req, res) => {
       const { prisonNumber } = req.params
-      // TODO: implement post action in next ticket
-      res.redirect(`/prisoners/${prisonNumber}/confirmation`)
+      const { username } = req.user
+      const { activeCaseLoadId } = res.locals.user
+      const data = await this.transfersService.getTransfer(activeCaseLoadId, prisonNumber)
+
+      raiseAnalyticsEvent(
+        'Add to the establishment roll',
+        'Confirmed transfer',
+        `AgencyId: ${activeCaseLoadId}, From: ${data.fromLocation}, Type: 'PRISON',`,
+        req.hostname
+      )
+
+      req.flash('prisoner', {
+        firstName: data.firstName,
+        lastName: data.lastName,
+      })
+
+      await this.transfersService.confirmTransfer(username, prisonNumber)
```

Comment:
maybe we should swap round the raiseAnalyticsEvent so it only happens in the confirmTransfer was successful?

Reply:
we should probably do the same for confirmArrival also if its been done that way

Reply:
ok